### ISSUE TITLE
specify format version specification

### ISSUE THREAD

**Issue by mtmorgan** [linnarsson-lab/loompy#34]
A formal version specification, bumped with each iteration and stored in the loom file, would be very useful.

**Comment by mtmorgan** [comment]
Thanks, I meant file specification version.

**Comment by slinnarsson** [comment]
The most sensible thing is probably to include a `loompy.__version__` attribute on the module, and let this track the latest tagged and released version. I've added this now to the master branch, and it works like this:

```python
>>> import loompy
>>> print(loompy.__version__)

2.0.6
```

It will be available in the next release.

**Comment by mtmorgan** [comment]
but this seems to version the data format on the software? Won't you be making software changes that don't alter the data format?

**Comment by slinnarsson** [comment]
Aha, ok got it. Both are probably useful. Let me think about it but probably it makes sense to make it an attribute on the root group.

For now, I will make the file spec version the same as the loompy version. They may later diverge, but with the caveat that we will not allow backwards-incompatible changes to the file spec. It should always be possible to open an old file with a new loompy library.

**Comment by slinnarsson** [comment]
### Experimental new file feature: file spec version stamp

Available in loompy 2.0.7.

`LOOM_SPEC_VERSION` HDF5 attribute on the root group. Use it like this:

```python
with loompy.connect(filename) as lc:
    if "LOOM_SPEC_VERSION" in lc.attrs:
        print("file version: " + lc.attrs.LOOM_SPEC_VERSION)
    else:
        print("file version: less than 2.0.1")
```

**Note**: this is experimental, and is a proposed new feature of the loom file format specification. As such, the attribute is not guaranteed to exist, and you must always check for it before trying to read it. If it doesn't exist, then the loom file spec version is assumed to be less than "2.0.1" by default.

**Note**: the LoomSpecVersion is not the same thing as the loompy package version.
